### ISSUE TITLE
EmbedMolecule with CoordMap Incorrectly translates final coords to the origin?

### ISSUE THREAD

**Issue by paconius** [rdkit/rdkit#6186]
**Describe the bug**
When EmbedMolecule is used with the coordMap option, the call appears to correctly perform the embedding, but then it seems to translate the molecule to the origin as a final step. Since I supplied explicit coordinates, it is unexpected that the final coordinates are centered around the origin instead of the supplied coordMap.

**To Reproduce**

```
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdChemReactions
from itertools import chain
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import Draw

#pyrrole with specific coordinates that are not centered around the origin
mol = Chem.MolFromMolBlock("""
  TESTER           3D

  5  5  0  0  0  0  0  0  0  0999 V2000
  -59.3708  -20.9320  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -60.7208  -20.4930  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -61.5558  -21.6420  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -60.7208  -22.7910  -51.5546 N   0  0  0  0  0  0  0  0  0  0  0  0
  -59.3708  -22.3520  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  1  5  2  0  0  0  0
  2  3  2  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
M  END
""")

#transform that ring expands the pyrrole ring into benzene
rxn = AllChem.ReactionFromSmarts('[o,s,nX3;r5;R1:1]:[ar5:2]>>[c:1]:[c]:[*:2]')
output = rxn.RunReactants((mol,))
prod = list(chain(*output))[0]

#peek at the product and notice that the unchanged atoms have the expected coords
#however the new C atom has coords at the origin (unwanted, but expected) -- we will fix  using EmbedMolecule()
prod
print(Chem.MolToMolBlock(prod))

#set the coord map using the coords from the starting material
cmap = {}
for i,atom in enumerate(prod.GetAtoms()):
    d = atom.GetPropsAsDict()
    react_atom_idx = d.get('react_atom_idx', None)
    if react_atom_idx:
        cmap[atom.GetIdx()] = mol.GetConformer().GetAtomPosition(react_atom_idx)

#print the coordMap and notice that the points correctly match the starting pyrrole
[(i,j.x,j.y,j.z) for i,j in cmap.items()]

#Embed the product and observe the final coordinates. The new embedding looks good, but is centered in the wrong spot
AllChem.EmbedMolecule(prod,coordMap=cmap)
print(Chem.MolToMolBlock(prod))
prod
```

Output Snippets:
```
>>> prod
---See first image pasted below the output box---
>>> print(Chem.MolToMolBlock(prod))
...
 6  6  0  0  0  0  0  0  0  0999 V2000
  -60.7208  -22.7910  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  -61.5558  -21.6420  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -59.3708  -22.3520  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -59.3708  -20.9320  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
  -60.7208  -20.4930  -51.5546 C   0  0  0  0  0  0  0  0  0  0  0  0
...
>>> [(i,j.x,j.y,j.z) for i,j in cmap.items()]
[(0, -60.7208, -22.791, -51.5546),
 (2, -61.5558, -21.642, -51.5546),
 (3, -59.3708, -22.352, -51.5546),
 (5, -60.7208, -20.493, -51.5546)]
>>>print(Chem.MolToMolBlock(prod))
...
 6  6  0  0  0  0  0  0  0  0999 V2000
   -0.8916    0.3801    0.0593 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.3282   -0.3931   -0.7829 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.1387   -0.6914   -0.6724 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0148    0.9505    0.8585 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2741    0.3817    0.7552 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.0992   -0.6277   -0.2176 C   0  0  0  0  0  0  0  0  0  0  0  0
...
>>>prod
---See second image pasted below the output box---
```
![image](https://user-images.githubusercontent.com/309160/224209731-29fb92ed-290d-4ab6-a44c-1293577e498d.png)
![image](https://user-images.githubusercontent.com/309160/224213825-2e2ceade-fc20-41bb-90e9-f49883616ef7.png)

**Expected behavior**
I expect the final coords to have the same _relative_ positions as what is produced by the above example, however I expect the molecule to be centered closer to (-60,-20,-50) than (0,0,0).

**Configuration (please complete the following information):**
 - RDKit version:  2022.03.1
 - OS: Windows 11
 - Python version (if relevant): 3.7.7
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge

**Comment by ptosco** [comment]
@paconius I believe this is the expected behavior.
This topic has come up previously on the mailing list and Greg has published a dedicated blog post:
http://rdkit.blogspot.com/2019/01/more-on-constrained-embedding.html
Check in particular the notebook cell `[9]` and the explanation above it.
For other use cases you might not be interested in the placement of the embedded molecule relative to the original `coordMap` coordinates, which is why the molecule is not realigned by default.
I would therefore close this bug report; please feel free to reopen if needed.

**Comment by paconius** [comment]
> This topic has come up previously on the mailing list and Greg has published a dedicated blog post:
> http://rdkit.blogspot.com/2019/01/more-on-constrained-embedding.html

Thank you @ptosco for posting that link. I think it is certainly relevant. The point about using AlignMol to cleanup after EmbedMolecule seems to be one that could work (although the extra MCS and Alignment are comparatively expensive time-wise). 

Even after reading Greg's blog post, I am still of the opinion that EmbedMolecule isn't working correctly. Greg's post seems to claim that due to the forecefield contraints set by EmbedMolecule, it isn't expected that the final embedding will match the provided coordMap *exactly*. This seems reasonable and expected given the method.
However, this cannot explain how a molecule that should be centered near [-60,-20,-50], ends up being centered around [0,0,0]. To my mind this can only be explained by a deliberate (and unwanted) translation to the origin. I'm not a C++ person, but within the C++ code for EmbedParameters I see references to "Center in Volume" -- which suggests that the code might indeed be centering the molecule and it might be using a false assumption that things should be centered around [0,0,0] -- since default conformer generation in RDKit produces [0,0,0] centered conformers. This assumption about how conformers are centered will not necessarily be true for confomations generated outside of RDKit.

TLDR: I beleive the EmbedMoelcule procedure makes a false assumption that conformers are centered around [0,0,0] and that this is a bug.
